skyportal/handlers/api/photometric_series.py

```diff
@@ -0,0 +1,1764 @@
+        Otherwise will return the list of ints.
+        In any case, it will append the user's
+        single user group ID to the list,
+        and return only unique values.
+    """
+    group_ids = data.pop("group_ids", [])
+    if isinstance(group_ids, str) and group_ids == "all":
+        public_group = session.scalars(
+            sa.select(Group).where(Group.name == cfg["misc.public_group_name"])
+        ).first()
+        group_ids = [public_group.id]
+    if isinstance(group_ids, (int, str)):
+        group_ids = [group_ids]
+    if not isinstance(group_ids, (list, tuple)):
+        raise ValidationError(
+            "Invalid group_ids parameter value. Must be a list of IDs "
+            "(integers) or the string 'all'."
+        )
+    for group_id in group_ids:
+        try:
+            group_id = int(group_id)
+        except TypeError:
+            raise ValidationError(
+                f"Invalid format for group id {group_id}, must be an integer."
+            )
+        group = session.scalars(Group.select(user).where(Group.id == group_id)).first()
+        if group is None:
+            raise ValidationError(f'Invalid group ID: {group_id}')
+
+    # always add the single user group
+    group_ids.append(user.single_user_group.id)
+    group_ids = list(set(group_ids))
+    return group_ids
+
+
+def get_stream_ids(data, user, session):
+    """
+    Get stream IDs from the request data.
+
+    Parameters
+    ----------
+    data: dict
+        Dictionary that can contain a "stream_ids" key.
+    user: skyportal.models.User
+        The user associated with the request.
+    session: sqlalchemy.orm.Session
+        The database session.
+
+    Returns
+    -------
+    stream_ids: list of int
+        Verifies each of the stream IDs in the input list
+        and returns the list of unique values.
+    """
+    stream_ids = data.pop("stream_ids", [])
+    if not isinstance(stream_ids, (list, tuple)):
+        raise ValidationError(
+            "Invalid stream_ids parameter value. Must be a list of IDs (integers)."
+        )
+    for stream_id in stream_ids:
+        try:
+            stream_id = int(stream_id)
+        except TypeError:
+            raise ValidationError(
+                f"Invalid format for stream id {stream_id}, must be an integer."
+            )
+        stream = session.scalars(
+            Stream.select(user).where(Stream.id == stream_id)
+        ).first()
+
+        if stream is None:
+            raise ValidationError(f'No stream with ID {stream_id}')
+
+    stream_ids = list(set(stream_ids))
+    return stream_ids
+
+
+def load_dataframe_from_bytestream(data):
+    """
+    Load the pandas data frame from a byte stream.
+    Will return a DataFrame and a metadata dictionary,
+    if such a dictionary is found in the data file's attributes.
+
+    ref: https://github.com/pandas-dev/pandas/issues/9246#issuecomment-74041497
+    """
+    with pd.HDFStore(
+        "data.h5",
```

Comment:
should be a temporary named file?

Reply:
I think in this case the filename doesn't make any difference.